pkg/mattermost/mattermost_v1beta.go

```diff
@@ -142,6 +170,70 @@ func GenerateIngressV1Beta(mattermost *mmv1beta.Mattermost) *networkingv1.Ingres
 	return ingress
 }
 
+func GenerateALBIngressClassV1Beta(mattermost *mmv1beta.Mattermost) *networkingv1.IngressClass {
+	ingressClass := &networkingv1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "alb",
```

Comment:
I took from the Spec on line 182, `ingress.k8s.aws/alb`. Im open to other suggestions though.

Reply:
I think the easiest way to move forward here is to use the name of the installation `mattermost.Name`, that way we know the ingress class belongs to a certain installation and we can treat that resource accordingly in the installations lifecycle. if we don't need it (because the spec now points to a custom one or because the ingress got disabled) we can just dispose of it but if we share the default ingress class between installations it would add complexity to the cleanup process, since we would need to check if there are other installations using it before trying its removal. Thoughts @gabrieljackson ?

Reply:
This is a good idea, Ive made the required changes.